### PR TITLE
Search result links

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #2611.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- search for something in the store
- right click an item and open it in a new tab
- the new tab should load correctly
- back on the search results click an item
- the item should load, but not reload the whole gui

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7756/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7757/

File: jujugui/static/gui/src/app/components/search-results/item/item.js [hunk 1]
```diff
@@ -171,7 +197,7 @@ YUI.add('search-results-item', function(Y) {
       @param {Object} e The click event.
     */
     _handleTagClick: function(tag, e) {
-      e.stopPropagation();
+      e.preventDefault();
```
Comment (Lukewh) [inline jujugui/static/gui/src/app/components/search-results/item/item.js]: `e` > `evt`

**Comment by Lukewh** [comment]
## QA OK

Great thanks @huwshimi

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7758/

File: jujugui/static/gui/src/app/components/search-results/_search-results.scss [hunk 2]
```diff
@@ -105,13 +105,29 @@
...
 
       @media only screen and (min-width : 768px) {
         padding: 8px 20px 4px;
       }
 
+      &-main-link {
+        position: absolute;
+        // Raise the main link above the children (but below the child links)
+        // so that it can be clicked on.
+        z-index: 1;
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/search-results/_search-results.scss]: Should this use the z-index function? (And the ones below)

Comment (huwshimi) [inline jujugui/static/gui/src/app/components/search-results/_search-results.scss]: The z-indexes here are scoped inside the charmbrowser so these will be fine.

**Comment by huwshimi** [comment]
Thanks @hatched and @Lukewh :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**